components/FilterForm/index.js

```diff
@@ -30,47 +49,65 @@ export default function FilterForm({ plants, onAddPreference }) {
   function handleSubmit(event) {
     event.preventDefault();
 
-    const newPreference = {
-      preferenceTitle: event.target.elements.title.value,
+    const { plantSize, sunlightRequirement, waterNeeds, optimalTemperature, petFriendly } = state;
+
+    const filterSettings = {
+      plantSize,
+      sunlightRequirement,
+      waterNeeds,
+      optimalTemperature,
+      petFriendly,
+    };
+
+    const preferenceData = {
+      preferenceTitle: state.preferenceTitle,
       preferencePlants: plants
         .filter(
           (plant) =>
-            filterPlantSize(plant.id, plantSize) &&
-            filterSunlightRequirement(plant.id, sunlightRequirement) &&
-            filterWaterNeeds(plant.id, waterNeeds) &&
-            filterOptimalTemperature(plant.id, optimalTemperature) &&
-            filterPetFriendly(plant.id, petFriendly)
+            filterPlantSize(plant.id, state.plantSize) &&
+            filterSunlightRequirement(plant.id, state.sunlightRequirement) &&
+            filterWaterNeeds(plant.id, state.waterNeeds) &&
+            filterOptimalTemperature(plant.id, state.optimalTemperature) &&
+            filterPetFriendly(plant.id,state.petFriendly)
         )
         .map((plant) => plant.id),
+      filterSettings,
     };
 
-    onAddPreference(newPreference);
-    event.target.elements.title.focus();
-    event.target.reset();
-  }
+    if (preferenceId) {
+      // Editing an existing preference
+      onEditPreference({ ...preferenceData, id: preferenceId });
+      router.push("/preferences");
+    } else {
+      // Adding a new preference
+      onAddPreference(preferenceData);
+      event.target.reset();
+    }
```

Comment:
I would suggest that the comments are not really needed as you already have really good function names. :)